### ISSUE TITLE
Online SGDClassifier with partial_fit

### ISSUE THREAD

**Issue by elinsky** [elinsky/market-microstructure-ml#27]
## Parent Epic
Part of #25 ML Prediction Pipeline

## Description

Create an online ML model that learns incrementally from streaming data using SGDClassifier.

### File: `src/model/classifier.py`

```python
from sklearn.linear_model import SGDClassifier
from sklearn.preprocessing import StandardScaler

class OnlineClassifier:
    def __init__(self, learning_rate: float = 0.01):
        self._model = SGDClassifier(
            loss='log_loss',
            learning_rate='constant',
            eta0=learning_rate,
            class_weight='balanced',
            warm_start=True,
        )
        self._scaler = StandardScaler()
        self._is_fitted = False
        self._sample_count = 0
        self._lock = threading.RLock()
    
    def partial_fit(self, features: np.ndarray, label: int) -> None:
        """Update model with single sample."""
        
    def predict_proba(self, features: np.ndarray) -> float:
        """Return probability of price change (class 1)."""
        
    @property
    def is_ready(self) -> bool:
        """True if model has seen enough samples to predict."""
```

## Features Used

From `FeatureSnapshot`:
- `spread_bps` - Spread in basis points
- `imbalance` - Bid-ask volume imbalance (-1 to 1)
- `depth` - Total order book depth
- `volatility` - Rolling price volatility

## Warm-up Period

Model needs minimum samples before predictions are meaningful:
- Minimum 100 samples of each class before `is_ready = True`
- Before ready, return 0.5 (no prediction)

## Acceptance Criteria

- [ ] SGDClassifier trains incrementally via `partial_fit()`
- [ ] Features are scaled before training/prediction
- [ ] Model handles class imbalance with weights
- [ ] `is_ready` property indicates when predictions are meaningful
- [ ] Thread-safe for concurrent training and prediction

**Comment by elinsky** [comment]
Completed in commit 731a985
